pylint/checkers/typecheck.py

```diff
@@ -1525,6 +1531,32 @@ def visit_subscript(self, node):
         if not supported_protocol(inferred):
             self.add_message(msg, args=node.value.as_string(), node=node.value)
 
+    @check_messages("dict-items-missing-iter")
+    def visit_for(self, node):
+        # not k, v
+        if not isinstance(node.target, astroid.node_classes.Tuple):
+            return
+
+        itr = node.iter
+
+        # not method
+        if not isinstance(itr, astroid.node_classes.Call):
```

Comment:
This suggests to me that it only looks for `Call` nodes as the iterable, is that the case? I think a simpler solution would be to:
- look for `Name`. This usually indicates that we tried to iterate a dictionary stashed in a variable
- Call`pylint.checkers.utils.safe_infer(iterable)` and check that the return value is a `Dict`
- If it is a Dict and if we have a tuple of two values in the `target`, then emit the error

This should catch most of the occurrences and it will also have less false positives.